app/services/plan_service/api/plans.rb

```diff
@@ -73,6 +74,31 @@ def get_trials(after:)
       Result::Success.new(PlanStore.get_trials(after: after))
     end
 
+    def get_external_service_link(marketplace_data)
+      Maybe(APP_CONFIG.external_plan_service_url)
+        .map { |external_plan_service_url|
+          marketplace_id = marketplace_data[:id]
+          Maybe(PlanStore.get_trial(community_id: marketplace_id))
+            .map { |trial_data|
+              trial_hash = CharmPlan.call(HashUtils.rename_keys({
+                id: :marketplace_plan_id,
+                community_id: :marketplace_id
+              }, trial_data))
+              payload = {
+                marketplace: marketplace_data,
+                initial_trial_data: trial_hash
+              }
+
+              secret = APP_CONFIG.external_plan_service_secret
+              url = external_plan_service_url + "/login"
+              token = JWTUtils.encode(payload, secret)
+              URLUtils.append_query_param(url, "token", token)
```

Comment:
Good point. 
(Result seems to be a bit overkill for fetching link for (currently) internal use, but it's in line with API practices)